### PR TITLE
Complete apache auth (SAML/SSO) flow with redirect page

### PR DESCRIPTION
**Pull request by mrow4a**
## Description
Without this change, server-side redirect to default page for Apache Auth (e.g. Shibboleth) resulted in session data missing on client side. Flow was incomplete and was missing final redirect page on client side to correctly handle redirect_url and persist the session data on the client.

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/5225
- Fixes https://github.com/owncloud/enterprise/issues/4712
- Related https://github.com/owncloud/core/pull/31054

## How Has This Been Tested?
- manually in internal oc2.lab test instance

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] Base code changes
- [x] Make sure to handle corner cases in login
- [x] Unit tests added
- [x] Acceptance tests added
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by mrow4a** [comment]
I did not add acceptance tests as part of this PR, as these need to be in the apps handling the backend, in here we can only cover with unit tests

**Comment by mrow4a** [comment]
@JanAckermann @DeepDiver1975 could you have a look here? target for this is [oc10.11](https://github.com/owncloud/enterprise/issues/5246)

File: core/Controller/LoginController.php [hunk 1]
```diff
@@ -122,10 +123,32 @@ public function logout() {
...
-		if (\OC_User::handleApacheAuth() || $this->userSession->isLoggedIn()) {
-			return new RedirectResponse($this->getDefaultUrl());
+		// check if there is apache auth backend available and try to obtain session,
+		// if apache backend not registered or failed to login, proceed with show login form
+		if ($this->handleApacheAuth()) {
+			// apache auth was completed server-side and there is active session,
+			// initiate login success redirect on client-side
+			//
+			// NOTE: as of https://github.com/owncloud/core/pull/31054 to allow alternative login methods
+			// one needs to setup https://doc.owncloud.com/server/10.10/admin_manual/enterprise/user_management/user_auth_shibboleth.html#other-login-mechanisms
```
Comment (mmattel) [inline core/Controller/LoginController.php:133]:
```suggestion
			// one needs to setup https://doc.owncloud.com/server/next/admin_manual/enterprise/user_management/user_auth_shibboleth.html#other-login-mechanisms
```

**Comment by mmattel** [comment]
I just made a small doc fix, unhide to see details.

**Comment by DeepDiver1975** [comment]
As discussed offline - as long as this works :+1:

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by phil-davis** [comment]
> @phil-davis I just added a comment text fix which resulted in `Build was killed` ... any CI hickup?

This PR was quite a lot of commits behind master. I rebased just now - let's see if CI is happier.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/36296/97/14
apiWebdavPreviews/previews.feature:210

That is failing in master today - issue https://github.com/owncloud/QA/issues/750 - it is being looked at...

**Comment by mrow4a** [comment]
@phil-davis btw, as mentioned in https://github.com/owncloud/enterprise/issues/5225#issuecomment-1167075255 and https://github.com/owncloud/enterprise/issues/5225#issuecomment-1166973577, as there is no acceptance test in user_shibolleth , we would need to make sure all works with SAML/SSO enabled in QA env.
